PyStemmusScope/soil_io.py

```diff
@@ -213,18 +214,132 @@ def prepare_soil_data(config):
     Args:
         config (dict): The PyStemmusScope configuration dictionary.
     """
+    loc, fmt = utils.check_location_fmt(config["Location"])
 
-    forcing_file = utils.get_forcing_file(config)
+    if fmt == "site":
+        forcing_file = utils.get_forcing_file(config)
+        # Data missing at ID-Pag site. See github.com/EcoExtreML/STEMMUS_SCOPE/issues/77
+        if config["Location"].startswith("ID"):
+            lat, lon = -1., 112.
+        else:
+            lat, lon = _retrieve_latlon(forcing_file)
+
+    elif fmt == "latlon":
+        lat = loc[0]
+        lon = loc[1]
 
-    # Data missing at ID-Pag site. See github.com/EcoExtreML/STEMMUS_SCOPE/issues/77
-    if config["Location"].startswith("ID"):
-        lat, lon = -1., 112.
     else:
-        lat, lon = _retrieve_latlon(forcing_file)
+        raise NotImplementedError
 
     matfiledata = _collect_soil_data(Path(config['SoilPropertyPath']), lat, lon)
 
     hdf5storage.savemat(
         Path(config["InputPath"]) / "soil_parameters.mat", mdict=matfiledata, appendmat=False,
     )
     utils.remove_dates_from_header(Path(config["InputPath"]) / "soil_parameters.mat")
+
+
+def prepare_soil_init(config):
+    """Function that prepares the soil inital conditions data for the STEMMUS_SCOPE
+    model. It parses the data for the input location, and writes a file that can be
+    easily read in by Matlab.
+
+    Args:
+        config (dict): The PyStemmusScope configuration dictionary.
+    """
+    loc, fmt = utils.check_location_fmt(config["Location"])
+
+    if fmt == "site":
+        matfiledata = _read_soil_initial_conditions_plumber2(
+            soil_init_path=Path(config['InitialConditionPath']),
+            sitename=loc,
+        )
+    elif fmt == "latlon":
+        matfiledata = _read_soil_initial_conditions_global(
+            global_path=Path(config['GlobalDataPath']),
```

Comment:
```suggestion
            global_path=Path(config['InitialConditionPath']),
```

Reply:
The keys in the config file should be the same for plumber2 data and global datasets. The value that is the path to the data can be different.

Reply:
I updated this for forcing & soil_init